samples/page-extension-react-router/src/index.js

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect } from 'react';
+import { render } from 'react-dom';
+import PropTypes from 'prop-types';
+import { Router, Switch, Route, Link } from 'react-router-dom';
+import { createMemoryHistory } from 'history';
+import { init, locations } from 'contentful-ui-extensions-sdk';
+import { Tab, Tabs, TabPanel, Heading, Button } from '@contentful/forma-36-react-components';
+import '@contentful/forma-36-react-components/dist/styles.css';
+import '@contentful/forma-36-fcss';
+import './index.css';
+
+function MainPage() {
+  return (
+    <TabPanel id="main" className="f36-margin-top--l">
+      Main page
+    </TabPanel>
+  );
+}
+
+function OtherPage() {
+  return (
+    <TabPanel id="other" className="f36-margin-top--l">
+      <div>Other page</div>
+      <Link to="/">Go back to main tab</Link>
+    </TabPanel>
+  );
+}
+
+function Page404() {
+  return <Heading>404</Heading>;
+}
+
+class PageExtension extends React.Component {
+  constructor(props) {
+    super(props);
+    this.history = createMemoryHistory({ initialEntries: [props.sdk.parameters.invocation.path] });
+    this.history.listen(location => {
+      this.props.sdk.navigator.openPageExtension({ path: location.pathname });
+    });
+  }
+
+  render = () => {
+    return (
+      <div className="f36-margin--l">
+        <Router history={this.history}>
+          <Tabs>
+            <Route
+              render={props => (
+                <>
+                  <Tab
+                    id="main"
+                    selected={props.location.pathname === '/'}
+                    onSelect={() => {
+                      props.history.push('/');
+                    }}>
+                    Main
+                  </Tab>
+                  <Tab
+                    id="other"
+                    selected={props.location.pathname === '/other'}
+                    onSelect={() => {
+                      props.history.push('/other');
+                    }}>
+                    Other
+                  </Tab>
+                </>
+              )}
+            />
+          </Tabs>
+          <Switch>
+            <Route path="/" exact render={() => <MainPage />} />
+            <Route path="/other" exact render={() => <OtherPage />} />
+            <Route render={() => <Page404 />} />
```

Comment:
Non-blocking: We could make this a little more concise by just passing the React component in the `render` prop.
For example:  `<Route render={Page404} />` since we don't pass router props to any of the above components in the routes.